docs/WritingTests.asciidoc

```diff
@@ -409,13 +407,75 @@ sub run {
     my $child_id = (keys %$children)[0];
 
     # this blocks until lock is available and then does nothing
-    mutex_unlock('child_reached_given_point', $child_id);
+    mutex_wait('child_reached_given_point', $child_id);
 
     # continue with the test
 }
 --------------------------------------------------------------------------------
 ====
 
+
+Mutexes are a way to wait for specific event from single job.
+When we need multiple jobs to reach required state we need to use barriers.
+
+Before first use barrier needs to be created with barrier_create with 2 parameters - name and count.
+Name behaves as ID (same as with mutexes), count is number of jobs needed to call barrier_wait to unlock barrier.
+
+There is optional barrier_wait parameter called check_dead_job.
+When used it will kill all jobs waiting in barrier_wait if one of cluster jobs die.
+
+It prevents waiting for state that will never be reached (and eventually die on job timeout).
+Should be set only on one of barrier_wait calls.
+
+Example is situation with 1 master and 3 worker jobs. We need to wait until 3 worker jobs perform initial setup.
+After that we can make cluster from them, but if one of them fails it makes no sense waiting.
+
+
+[caption="Example of barriers: "]
+.Check for dead jobs while waiting for barrier
+====
+[source,perl]
+--------------------------------------------------------------------------------
+use lockapi;
+
+# In main.pm
+barrier_create('NODES_CONFIGURED', 4);
+
+# On master job
+sub run {
+    assert_screen 'login', 300;
+
+    # Master is ready, waiting while workers are configured
+    barrier_wait 'NODES_CONFIGURED';
```

Comment:
I don't see the use a `check_dead_job`, is that not the goal of this example? Or I misunderstood something maybe :-)